### ISSUE TITLE
Responsive Accordion Deep Linking

### ISSUE THREAD

**Issue by mediabeastnz** [zurb/foundation-sites#9494]
The new responsive accordion doesn't seem to play nice with deep linking. 

Events on either Tabs or Accordion only fire when the media query event is triggered on the new plugin.

It would be ideal if these events worked as expected with out having to resize or if deep linking was built into ResponsiveAccordion.

**Comment by kball** [comment]
interesting, thanks for trying out 6.3 ahead of the release and filing this!  Will take a look shortly - don't suppose you have an example already coded up I can use to debug?

**Comment by mediabeastnz** [comment]
No unfortunately I don't have one that's easy to share (lots of SilverStripe CMS templating logic throughout).

I just copied the example from the 6.3 docs and attempted a few different approaches to enable deep linking as well as attempting to trigger an even when a hash is in the URL but that event only fired when the browser was resized.

**Comment by kball** [comment]
Oof - accordion doesn't have deeplink support at this moment, so a full solution will require implementing that.

However, I found that if I add `data-deep-link="true"` to the element that has the `data-responsive-accordion-tabs` component, it appears to work for me when the accordion-tabs is in the tabs form... both on changing tabs and then loading a page with the hash url.  The caveat is that when if the component is in accordion form either when navigating it or when coming in via a link, it doesn't work.

@mediabeastnz can you confirm?  Or does even this case not work for you?

**Comment by mediabeastnz** [comment]
It doesn't seem to work - although if i have:
```html
<li class="accordion-item" data-accordion-item>
    <a href="#" id="tab3" class="accordion-title">Accordion 3</a>
</li>
```
and then link that page with the hash #tab3 it highlights the tab but doesn't open the content nor does it do anything when in accordion mode.

**Comment by kball** [comment]
Try adding the ids to the hrefs... I'll see about updating the docs, but you can take a look at the testcase I was playing around with here: https://github.com/zurb/foundation-sites/commit/93c39c7e679d02885078899c5e3aa3c2bd46c35b

**Comment by mediabeastnz** [comment]
Ahh I see I think we had ID's as well as href's.

So from this...

1. Deep linking is only supported on tabs (if you use accordion for mobile a deep link wont work).
2. Deep linking seems to be stripping the URL back to just the hash e.g. `/my-page#myhash` becomes `/#myhash`

At least we're aware of the limitations of the current version of responsive tabs/accordion :)

**Comment by kball** [comment]
number 1 definitely correct... number2 shouldn't (doesn't in the testcase I posted a link to)... can you post your updated code so I can take a look?

**Comment by mediabeastnz** [comment]
I just copied your code from [93c39c7](https://github.com/zurb/foundation-sites/commit/93c39c7e679d02885078899c5e3aa3c2bd46c35b)

Only difference is I think i'm using 6.3 RC-1 - don't think that would make a difference though.

It seems to load the correct tab and then 1 second later strip the URL. Weird huh?

**Comment by kball** [comment]
Bizarre.  I'm at the end of my day but I'll see if I can dig in more tomorrow, as well as updating the docs with this.  If you have a chance, would love to see if you also see this behavior with RC3...

Oh, one last thing to double check.  Can you post what browser you're testing with? 

Thanks for helping track this down

**Comment by mediabeastnz** [comment]
Ok I did some digging...
Seems to happen on all recent version just FYI.

If deep linking is enabled there is a statement which does some browser history updates.
[This line here](https://github.com/zurb/foundation-sites/blob/develop/js/foundation.tabs.js#L246) seems to be rewriting the URL to be just `/#myanchor` 

The history state seems to be undefined at that stage which might be the issue. If history isn't turned on it should just skip all that logic? Easy fix to just remove it?

Tested on Chrome and Safari ;)

**Comment by kball** [comment]
hmm... I can reproduce the behavior if I change the link to be `/#anchor` instead of `#anchor`, but I think that is an invalid anchor... can you confirm if your links have a `/` in the front of them?

**Comment by ncoden** [comment]
@mediabeastnz Is it still an issue ?

**Comment by jwatkin** [comment]
Was this supposed to have been fixed in ZURB Foundation 6.4.3? Deep linking responsive accordion tabs isn't working for me when using the 6.4.3 template located here:
[https://github.com/zurb/foundation-zurb-template](https://github.com/zurb/foundation-zurb-template)

For my test, I'm using this code, which is based on examples from the doc:

```
<div class="row">
    <div class="columns">
      <h2>Responsive Accordion Tabs</h2>
      <p>Either the <a href="//foundation.zurb.com/sites/docs/accordion.html">Accordion Markup</a> or the <a href="//foundation.zurb.com/sites/docs/tabs.html">Tabs Markup</a> can be used to responsively switch between the two components at different breakpoints.</p>
    </div>
  </div>
  
  <div class="row">
    <div class="columns">
      <h4>Accordion HTML Markup</h4>
      <div class="secondary callout">
        <p>Once you put it all together, here's what you get!<br /> Resize your browser window to see the toggle happen.<br>I am an `Accordion on small and large` but I am `Tabs on medium`</p>
      </div>
    </div>
  </div>
  
  <div class="row">
    <div class="columns">
      <ul class="accordion" data-responsive-accordion-tabs="accordion medium-tabs large-accordion" data-deep-link="true" data-update-history="true" data-deep-link-smudge="true" data-deep-link-smudge="500" id="deeplinked-accordion">
        <li class="accordion-item" data-accordion-item>
          <a href="#deeplink1" class="accordion-title">Accordion 1</a>
          <div class="accordion-content" data-tab-content id="deeplink1">
            <p>Panel 1. Lorem ipsum dolor</p>
            <a href="#">Nowhere to Go</a>
          </div>
        </li>
        <li class="accordion-item" data-accordion-item>
          <a href="#deeplink2" class="accordion-title">Accordion 2</a>
          <div class="accordion-content" data-tab-content id="deeplink2">
            <textarea></textarea>
            <button class="button">I do nothing!</button>
          </div>
        </li>
        <li class="accordion-item" data-accordion-item>
          <a href="#deeplink3" class="accordion-title">Accordion 3</a>
          <div class="accordion-content" data-tab-content id="deeplink3">
            Pick a date!
            <input type="date"></input>
          </div>
        </li>
      </ul>
    </div>
  </div>
```

When I open some accordions and try to resize the browser window a few times, the accordions don't open anymore and sometimes it won't switch to the tab view. I'm getting these JavaScript errors:

![image](https://user-images.githubusercontent.com/2364100/32205143-d9839ad4-bdaa-11e7-80f5-ea7a9a589e3c.png)

If I remove the deep-linking modifications from markup above, the responsive accordion tabs will work fine.

Thank you.

Jim

**Comment by jwatkin** [comment]
Someone named Chris posted on the ZURB Forum a workaround for my previous GitHub comment in this issue. Oddly, their comment is now missing from the ZURB Forum. The workaround is to remove the "smudge" attributes from the data-responsive-accordion-tabs. Thank you Chris.
